lib/rubocop/cop/mixin/surrounding_space.rb

```diff
@@ -107,9 +107,9 @@ def empty_offense(node, range, message, command)
         end
       end
 
-      def empty_brackets?(left_bracket_token, right_bracket_token)
-        left_index = processed_source.tokens.index(left_bracket_token)
-        right_index = processed_source.tokens.index(right_bracket_token)
+      def empty_brackets?(tokens, left_bracket_token, right_bracket_token)
```

Comment:
Note that adding `tokens` is potentially a breaking changes for users of `SurroundingSpace` mixin. As an alternative, we could pass tokens as an optional argument and default to the slow version `processed_source.tokens`.